crowbar_framework/lib/crowbar/backup/restore.rb

```diff
@@ -268,10 +268,21 @@ def run_installer
       def restore_database
         Rails.logger.debug "Restoring Crowbar database"
         begin
+          Crowbar::Migrate.migrate!
+        rescue SQLite3::SQLException => e
+          Rails.logger.debug "Failed to migrate database: #{e}"
+          set_failed
+          @status[:restore_database] = {
+            status: :not_acceptable,
+            msg: I18n.t("backups.index.migrate_database_failed")
+          }
+        end
+
+        begin
           SerializationHelper::Base.new(YamlDb::Helper).load(
```

Comment:
this will lead to some problems at Exception handling, as discussed here https://github.com/crowbar/crowbar-core/pull/307#discussion_r52455950